diag_manager/fms_diag_field_object.F90

```diff
@@ -65,7 +65,12 @@ module fms_diag_field_object_mod
      integer, allocatable, private                    :: area, volume      !< The Area and Volume
      class(*), allocatable, private                   :: missing_value     !< The missing fill value
      class(*), allocatable, private                   :: data_RANGE(:)     !< The range of the variable data
-    contains
+     class(*), allocatable, dimension(:,:,:,:), private :: data_buffer     !< Buffer for field data
```

Comment:
Are you using this instead of a buffer object, or is this just for the initial call? I think in the original design the buffer id is stored here and then the actual buffers are in an array in the diag object, not sure if that makes more sense.

Reply:
This buffers the data from `accept_data` to be used in `fms_diag_send_complete`.  When the math functions are called, the buffer object buffers are filled in.  There are 2 possibilities:
1. Accept data is called and the math functions are performed in accept data.  In this case, this buffer is not allocated or used.
2. Accept data is called and the only thing that happens is this data buffer is allocated and the data is stored here.  The math functions will be done in fms_diag_send_complete.  This is a big change from how `send_data` in the current diag manager works.  

Why do this at all?  This will 
1. Make accept_data thread safe (which is why there is a check for threading).
2. Allow us to test/use openmp/`do concurrent` for the math part of the diag manager.
3. Provide offloading for the most computationally expensive part of the diag manager in the future.